runtime/lua/vim/lsp/util.lua

```diff
@@ -1622,6 +1617,22 @@ function M.open_floating_preview(contents, syntax, opts)
     api.nvim_buf_set_var(bufnr, 'lsp_floating_preview', floating_winnr)
   end
 
+  local augroup_name = ('closing_floating_preview_%d'):format(floating_winnr)
+  local ok =
+    pcall(api.nvim_get_autocmds, { group = augroup_name, pattern = tostring(floating_winnr) })
+  if not ok then
+    api.nvim_create_autocmd('WinClosed', {
+      group = api.nvim_create_augroup(augroup_name, {}),
+      pattern = tostring(floating_winnr),
+      callback = function()
+        if api.nvim_buf_is_valid(bufnr) then
+          vim.b[bufnr].lsp_floating_preview = nil
+        end
+        api.nvim_del_augroup_by_name(augroup_name)
```

Comment:
Why not just `return true` here? Then maybe the `augroup_name` isn't needed, also?